app/models/workflow_execution/query.rb

```diff
@@ -0,0 +1,202 @@
+# frozen_string_literal: true
+
+# Model to represent workflow execution search form
+# Provides advanced search capabilities for filtering workflow executions
+# Supports groups, conditions, operators, basic search, and pagination
+class WorkflowExecution::Query # rubocop:disable Style/ClassAndModuleChildren, Metrics/ClassLength
+  include ActiveModel::Model
+  include ActiveModel::Attributes
+  include Pagy::Backend
+  include AdvancedSearchable
+  include AdvancedSearchConditions
+
+  ResultTypeError = Class.new(StandardError)
+
+  ALLOWED_SORT_COLUMNS = %w[id name run_id state created_at updated_at].freeze
+
+  attribute :column, :string
+  attribute :direction, :string
+  attribute :name_or_id_cont, :string
+  attribute :name_or_id_in, default: -> { [] }
+  attribute :namespace_ids, default: -> { [] }
+  attribute :groups, default: lambda {
+    [WorkflowExecution::SearchGroup.new(
+      conditions: [WorkflowExecution::SearchCondition.new(field: '', operator: '', value: '')]
+    )]
+  }
+  attribute :sort, :string, default: 'updated_at desc'
+  attribute :advanced_query, :boolean, default: false
+
+  validates :direction, inclusion: { in: %w[asc desc] }
+  validates :namespace_ids, length: { minimum: 1 }
+  validates :column, inclusion: {
+    in: lambda { |record|
+      ALLOWED_SORT_COLUMNS + [record.column].select { |c| c&.start_with?('metadata.') }
+    }
+  }
+  validates_with WorkflowExecutionAdvancedSearchGroupValidator
+
+  def initialize(attributes = {}, scope: WorkflowExecution, **kwargs)
+    attributes = if attributes.present?
+                   attributes.merge(kwargs)
+                 else
+                   kwargs
+                 end
+
+    super(attributes)
+    @scope = scope
+    self.sort = sort
+    self.advanced_query = advanced_query?
+    self.groups = groups
+  end
+
+  def sort=(value)
+    super
+    # use rpartition to split on the first space encountered from the right side
+    # this allows us to sort by metadata fields which contain spaces
+    sort_value = sort.presence || 'updated_at desc'
+    column, _space, direction = sort_value.rpartition(' ')
+
+    # Fallback to default if column is empty (e.g., if sort was just "desc")
+    if column.blank?
+      column = 'updated_at'
+      direction = direction.presence || 'desc'
+    end
+
+    column = column.gsub('metadata_', 'metadata.') if column.match?(/metadata_/)
+    assign_attributes(column:, direction:)
+  end
+
+  def results(**results_arguments)
```

Comment:
Same comment as above for this method, also for the pagy_results/non_pagy_results methods below